.github/workflows/run-test-suite.yml

```diff
@@ -112,3 +113,153 @@ jobs:
+    # Important! Must install our development version of cfunits to test:
+    # pip install -e .
+    - name: install cfunits (ubuntu 18.04)
+      if: matrix.os == 'ubuntu-18.04'
+      run: pip3 install -e .
+    - name: install cfunits (ubuntu 20.04)
+      if: matrix.os == 'ubuntu-20.04'
+      # https://github.com/pypa/pip/issues/7953
+      run: pip3 install --user --no-use-pep517 -e .
+    - name: python3 -c "import cfunits; print(cfunits.__file__)"
+      run: |
+        mkdir foo
+        cd foo
+        python3 -c "import cfunits; print(cfunits.__file__)"
+    # Finally run the test suite and generate a coverage report!
+    - name: Run test suite and generate a coverage report
+      run: |
+        cd cfunits/test
+        ./run_tests_and_coverage --nohtml
+
+  run-test-suite-job_i386_debian:
+    name: run test suite on debian latest (32 bit)
+
+    # https://github.com/daniel-mohr/i386_debian_latest
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: create test script for docker
+        run: |
+          echo "#!/bin/sh -l" > run_test_suite
+          echo "set -e -v" >> run_test_suite
+          echo "echo 'install dependencies (debian packages)'" >> run_test_suite
+          echo "DEBIAN_FRONTEND=noninteractive apt install -q -y lsb-release python3-setuptools python3-wheel python3-pip libudunits2-0 python3-pycodestyle python3-coverage python3-numpy" >> run_test_suite
+          echo "echo 'check environment'" >> run_test_suite
+          echo "uname -a" >> run_test_suite
+          echo "dpkg --print-architecture" >> run_test_suite
+          echo "lsb_release -a" >> run_test_suite
+          echo "lscpu" >> run_test_suite
+          echo "python3 -V" >> run_test_suite
+          echo "pip3 install --user --no-use-pep517 -e ." >> run_test_suite
+          echo "mkdir foo" >> run_test_suite
+          echo "(cd foo && python3 -c 'import cfunits; print(cfunits.__file__)')" >> run_test_suite
+          echo "echo 'Run test suite and generate a coverage report'" >> run_test_suite
+          echo "cd cfunits/test" >> run_test_suite
+          echo "./run_tests_and_coverage --nohtml" >> run_test_suite
+          chmod +x run_test_suite
+      - name: cat test script
+        run: cat run_test_suite
+      - name: run test suite in docker image
+        uses: daniel-mohr/i386_debian_latest@v0
+        with:
+          cmdfile: ./run_test_suite
+
+  run-test-suite-job_windows:
+    name: run test suite on windows using Miniconda for package management
+
+    # Set-up the build matrix.
+    strategy:
+      # due to error in cfunits we disable fail-fast to check CI configuration
+      fail-fast: false
+      matrix:
+        # https://github.com/actions/virtual-environments/blob/main/images/win/Windows2016-Readme.md
+        # https://github.com/actions/virtual-environments/blob/main/images/win/Windows2019-Readme.md
+        os: [windows-2016, windows-2019]
```

Comment:
A warning has emerged for the Windows jobs, namely that:

> The windows-2016 environment is deprecated and will be removed on March 15, 2022. Migrate to windows-latest instead. For more details see https://github.com/actions/virtual-environments/issues/4312

and I notice that [the Actions Team are going to temporarily fail jobs using Windows 2016 to raise awareness of this, including tomorrow evening](https://github.com/actions/virtual-environments/issues/4312#issue-1030757007), so to address this before the deprecation, it would be good to change this `os` specification to just `windows-latest`, as they suggest.